homeassistant/components/sensor/nsw_fuel_station.py

```diff
@@ -0,0 +1,110 @@
+"""
+Component to display the current fuel prices at a NSW fuel station.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/nsw_fuel_station/
+"""
+import datetime
+from typing import Optional
+import voluptuous as vol
+
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nsw-fuel-api-client==1.0.7']
+
+CONF_STATION_ID = 'station_id'
+CONF_STATION_NAME = 'station_name'
+CONF_FUEL_TYPES = 'fuel_types'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.positive_int,
+    vol.Required(CONF_STATION_NAME): cv.string,
+    vol.Required(CONF_FUEL_TYPES, default=[]): vol.All(
```

Comment:
if it is required there is no point in having a default, maybe make it option and have the default a non empty list ?

Reply:
Great suggestion - I have changed the default to now be the full set of available fuel types and have updated the platform setup to only create sensors where there is an available fuel type to match in the station data (some stations don't supply every type of fuel)